### PR TITLE
EOA and Decompressor

### PR DESCRIPTION
**Pull request by K-Ho**
## Questions
- 
-
-

### Fixes
- Fixes # [Link to Issue]

## Contributing Agreement

- [X] I have read and understood the [Optimism Contributing Guide and Code of Conduct](https://github.com/ethereum-optimism/optimism-monorepo/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

**Comment by karlfloersch** [comment]
Still a WIP but fixing the broken tests

File: contracts/optimistic-ethereum/OVM/precompiles/OVM_SequencerEntrypoint.sol [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    }
+
+
+    /*********************
+     * Fallback Function *
+     *********************/
+
+    /**
+     * Uses a custom "compressed" format to save on calldata gas:
+     * calldata[00:01]: transaction type (0 == EIP 155, 2 == Eth Sign Message)
```
Comment (tynes) [inline contracts/optimistic-ethereum/OVM/precompiles/OVM_SequencerEntrypoint.sol:30]: The enum should be defined as 0 and 1 right?

File: contracts/optimistic-ethereum/OVM/precompiles/OVM_SequencerEntrypoint.sol [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+     */
+    function _getTransactionType(
+        uint8 _transactionType
+    )
+        internal
+        returns (
+            TransactionType
+        )
+    {
+        if (_transactionType == 0) {
```
Comment (kfichter) [inline contracts/optimistic-ethereum/OVM/precompiles/OVM_SequencerEntrypoint.sol:108]: Request to add quick comment here to explain why we're using 0 and 2.

**Review by kfichter — APPROVED** [review]
Ok LGTM.

**Comment by kfichter** [comment]
Resolved merge conflicts.

**Comment by kfichter** [comment]
Resolved merge conflicts.

**Pull request merged**